bug: The transformer-attributify-jsx does not work correctly

In the react project using unocss and transformerAttributifyJsx, I found that transformerAttributifyJsx plugin is unable to correctly handle the style property on the label or three expression, so I added some style property and three expression test in transformerAttributifyJsx plugin, find test case can't pass. I think this is a bug appeared on the regular expression matching rules (**attributeRE**).


### Unit Test
```ts
import MagicString from 'magic-string'
import presetAttributify from '@unocss/preset-attributify'
import presetUno from '@unocss/preset-uno'
import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import transformerAttributifyJsx from '../packages/transformer-attributify-jsx/src'
describe('transformerAttributifyJs', () => {
  const originalCode = `
<div h-full text-center flex select-none className={red ? 'text-red': 'text-green'}>
  <input value={ target ? '10px': '20px' } style={{ height: '100px' }}>
</div>
  `.trim()

  const uno = createGenerator({
    presets: [
      presetUno(),
      presetAttributify(),
    ],
  })

  test('transform', async () => {
    const code = new MagicString(originalCode)
    await transformerAttributifyJsx().transform(code, 'app.tsx', { uno, tokens: new Set() } as any)

    expect(code.toString()).toMatchInlineSnapshot(`
      "<div h-full=\\"\\" text-center=\\"\\" flex=\\"\\" select-none=\\"\\" className={red ? 'text-red': 'text-green'}>
          <input value={ target ? '10px' : '20px' } style={{ height: '100px' }}>
      </div>
    `)
  })
})
```

### Test Result
```bash
test/transformer-attributify-jsx.test.ts > transformerAttributifyJs > transform
 ❯ Proxy.methodWrapper node_modules/.pnpm/chai@4.3.6/node_modules/chai/lib/chai/utils/addMethod.js:57:25
AssertionError: Snapshot `transformerAttributifyJs > transform 1` mismatched
 ❯ test/transformer-attributify-jsx.test.ts:25:28
     23|     await transformerAttributifyJsx().transform(code, 'app.tsx', { uno, tokens: new Set() } as any)
     24| 
     25|     expect(code.toString()).toMatchInlineSnapshot(`
       |                            ^
     26|       "<div h-full=\\"\\" text-center=\\"\\" flex=\\"\\" select-none=\\"\\" className={red ? 'text-red': 'text-green'}>
     27|           <input value={ target ? '10px' : '20px' } style={{ height: '100px' }}>

  - Expected  - 2
  + Received  + 2

    "<div h-full=\\"\\" text-center=\\"\\" flex=\\"\\" select-none=\\"\\" className={red ? 'text-red': 'text-green'}>
  -     <input value={ target ? '10px' : '20px' } style={{ height: '100px' }}>
  - </div>"
  +   <input value={ target ? '10px=\\"\\"': '20px=\\"\\"' } style={{ height: '100px=\\"\\"' }}>
  + </div>""

 Snapshots  1 failed
```

--- a/packages/transformer-attributify-jsx/src/index.ts
+++ b/packages/transformer-attributify-jsx/src/index.ts
@@ -40,6 +40,7 @@ export interface TransformerAttributifyJsxOptions {
 const elementRE = /<!--[\s\S]*?-->|<(\/?)([a-zA-Z][-.:0-9_a-zA-Z]*)((?:\s+[^>]*?(?:(?:'[^']*')|(?:"[^"]*"))?)*)\s*(\/?)>/gs
 const attributeRE = /([a-zA-Z()#][\[?a-zA-Z0-9-_:()#%\]?]*)(?:\s*=\s*((?:'[^']*')|(?:"[^"]*")|\S+))?/g
 const classFilterRE = /(className|class)\s*=\s*\{[^\}]*\}/i
+const curlybraceRE = /\{?.+\}/g
 
 export default function transformerAttributifyJsx(options: TransformerAttributifyJsxOptions = {}): SourceCodeTransformer {
   const {
@@ -78,7 +79,8 @@ export default function transformerAttributifyJsx(options: TransformerAttributif
         let attributifyPart = item[3]
         if (classNamePart)
           attributifyPart = item[3].replace(classFilterRE, ' '.repeat(classNamePart[0].length))
-
+        if (curlybraceRE.test(attributifyPart))
+          continue
         for (const attr of attributifyPart.matchAll(attributeRE)) {
           const matchedRule = attr[0].replace(/\:/i, '-')
           if (matchedRule.includes('=') || isBlocked(matchedRule))

--- a/packages/transformer-attributify-jsx/src/index.ts
+++ b/packages/transformer-attributify-jsx/src/index.ts
@@ -40,7 +40,7 @@ export interface TransformerAttributifyJsxOptions {
 const elementRE = /<!--[\s\S]*?-->|<(\/?)([a-zA-Z][-.:0-9_a-zA-Z]*)((?:\s+[^>]*?(?:(?:'[^']*')|(?:"[^"]*"))?)*)\s*(\/?)>/gs
 const attributeRE = /([a-zA-Z()#][\[?a-zA-Z0-9-_:()#%\]?]*)(?:\s*=\s*((?:'[^']*')|(?:"[^"]*")|\S+))?/g
 const classFilterRE = /(className|class)\s*=\s*\{[^\}]*\}/i
-const curlybraceRE = /\{?.+\}/g
+const curlybraceRE = /\{.+\}/g
 
 export default function transformerAttributifyJsx(options: TransformerAttributifyJsxOptions = {}): SourceCodeTransformer {
   const {

--- a/test/transformer-attributify-jsx.test.ts
+++ b/test/transformer-attributify-jsx.test.ts
@@ -78,6 +78,7 @@ describe('transformerAttributifyJs', () => {
 
     expect(code.toString()).toMatchInlineSnapshot(`
       "<div h-full=\\"\\" text-center=\\"\\" flex select-none=\\"\\" className={red ? 'text-red': 'text-green'}>
+        <input value={ target ? '10px' : '20px'} style={{ height: '100px' }}>
         <div ma=\\"\\">
           <div text-5xl=\\"\\" fw100=\\"\\" animate-bounce-alt=\\"\\" animate-count-infinite=\\"\\" animate-duration-1s=\\"\\">
             unocss

--- a/test/transformer-attributify-jsx.test.ts
+++ b/test/transformer-attributify-jsx.test.ts
@@ -8,6 +8,7 @@ import transformerAttributifyJsx from '../packages/transformer-attributify-jsx/s
 describe('transformerAttributifyJs', () => {
   const originalCode = `
 <div h-full text-center flex select-none className={red ? 'text-red': 'text-green'}>
+  <input value={ target ? '10px' : '20px'} style={{ height: '100px' }}>
   <div ma>
     <div text-5xl fw100 animate-bounce-alt animate-count-infinite animate-duration-1s>
       unocss
@@ -43,6 +44,7 @@ describe('transformerAttributifyJs', () => {
 
     expect(code.toString()).toMatchInlineSnapshot(`
       "<div h-full=\\"\\" text-center=\\"\\" flex=\\"\\" select-none=\\"\\" className={red ? 'text-red': 'text-green'}>
+        <input value={ target ? '10px' : '20px'} style={{ height: '100px' }}>
         <div ma=\\"\\">
           <div text-5xl=\\"\\" fw100=\\"\\" animate-bounce-alt=\\"\\" animate-count-infinite=\\"\\" animate-duration-1s=\\"\\">
             unocss

